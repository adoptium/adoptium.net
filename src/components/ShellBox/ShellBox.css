.shell-box {
  --mono: consolas, monaco, "Andale Mono", "Ubuntu Mono", monospace;

  border-radius: 0.4rem;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  font-family: var(--mono);
  padding: 0 0 1.8rem 1.6rem;
  position: relative;
  overflow: visible;
}

.shell-box code {
  color: #ed5393;
  font-family: inherit;
  line-height: 30px;
  overflow-x: hidden;
  position: absolute;
  top: 30px;
  width: calc(100% - 20px);
  text-align: left;
}

.shell-box code:hover {
  overflow-x: auto;
}

.shell-box code::-webkit-scrollbar {
  height: 0.5em;
}

.shell-box code::-webkit-scrollbar,
.shell-box code::-webkit-scrollbar-thumb {
  border-radius: 4px;
  overflow: visible;
}

.shell-box code::-webkit-scrollbar-thumb {
  background: rgba(0, 0, 0, 20%);
}

.shell-box code > span.install-text-command {
  color: #fff;
}

.shell-box code > span.function {
  color: #d07912;
}

.shell-box .shell-box-top {
  display: inherit;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 0.8rem;
}

.shell-box .shell-box-top span,
.shell-box .shell-box-top button {
  align-items: center;
  display: inherit;
  font-size: 1.6rem;
  height: 23px;
  justify-content: center;
  width: 86px;
}

.shell-box .shell-box-top span {
  background-color: #d9e1e4;
  border-radius: 0 0.3rem 0.3rem;
  color: #2c3437;
  margin-left: 1.6rem;
}

.shell-box .shell-box-top button {
  background-color: #c5e5b4;
  border-radius: 0 0.3rem 0.3rem;
  border-width: 0;
}

.shell-box .shell-box-top button i {
  padding: 0;
}
