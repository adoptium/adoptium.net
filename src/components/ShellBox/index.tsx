import React from "react"

import "./ShellBox.css"

interface Props {
  children: React.ReactNode
  textToCopy?: string
}

const ShellBox = ({ children, textToCopy }: Props): React.ReactNode => {
  const handleCopy = () => {
    if (textToCopy) {
      navigator.clipboard.writeText(textToCopy);
    }
  };

  return (
    <pre className="shell-box no-highlight">
      <div className="shell-box-top">
        {textToCopy && (
          <button onClick={handleCopy} aria-label="Copy to clipboard">
            <i className="fa fa-copy" />
          </button>
        )}
      </div>
      <code className="shell-box-code">{children}</code>
    </pre>
  )
}

export default ShellBox
