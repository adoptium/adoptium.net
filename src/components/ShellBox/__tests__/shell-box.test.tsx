import React from "react"

import { render } from "@testing-library/react"
import { afterEach, describe, expect, it, vi } from "vitest"

import ShellBox from "../index"

Object.assign(navigator, {
  clipboard: {
    writeText: vi.fn(),
  },
})

const navigatorClipboardSpy = vi.spyOn(navigator.clipboard, "writeText")

afterEach(() => {
  vi.clearAllMocks()
})

describe("ShellBox component", (): void => {
  it("renders correctly", (): void => {
    const textToCopy = "text to be copy"
    const { container } = render(
      <ShellBox textToCopy={textToCopy}>mock-children-code</ShellBox>,
    )
    expect(container).toMatchSnapshot()
  })

  it("can copy text to clipboard", async () => {
    const textToCopy = "text to be copy"
    const { getByRole } = render(
      <ShellBox textToCopy={textToCopy}>mock-children-code</ShellBox>,
    )

    // Simulate clicking the copy button
    const copyButton = getByRole('button');
    copyButton.click();

    // Verify the clipboard writeText was called with the correct text
    expect(navigatorClipboardSpy).toHaveBeenCalledWith(textToCopy);
  })
})
