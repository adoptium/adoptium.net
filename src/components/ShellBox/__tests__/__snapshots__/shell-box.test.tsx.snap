// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShellBox component > renders correctly 1`] = `
<div>
  <pre
    class="shell-box no-highlight"
  >
    <div
      class="shell-box-top"
    />
    <code
      class="shell-box-code"
    >
      mock-children-code
    </code>
  </pre>
</div>
`;
